benches/oneshot/ipa.rs

```diff
@@ -3,32 +3,45 @@ use raw_ipa::{
     error::Error,
     test_fixture::{
         generate_random_user_records_in_reverse_chronological_order, test_ipa,
-        update_expected_output_for_user, IpaSecurityModel, TestWorld,
+        update_expected_output_for_user, IpaSecurityModel, TestWorld, TestWorldConfig,
     },
 };
+use std::num::NonZeroUsize;
 
 #[tokio::main(flavor = "multi_thread", worker_threads = 3)]
 async fn main() -> Result<(), Error> {
     const MAX_BREAKDOWN_KEY: usize = 16;
     const MAX_TRIGGER_VALUE: u32 = 5;
-    const NUM_USERS: usize = 8;
-    const MAX_RECORDS_PER_USER: usize = 8;
+    const MAX_QUERY_SIZE: usize = 100;
+    const NUM_USERS: usize = 30;
+    const MAX_RECORDS_PER_USER: usize = 10;
+
+    let mut config = TestWorldConfig::default();
+    config.gateway_config.send_buffer_config.items_in_batch = NonZeroUsize::new(1).unwrap();
+    config.gateway_config.send_buffer_config.batch_count = NonZeroUsize::new(1024).unwrap();
 
     let random_seed = thread_rng().gen();
     println!("Using random seed: {random_seed}");
     let mut rng = StdRng::seed_from_u64(random_seed);
 
+    let mut total_count = 0;
+
     let mut random_user_records = Vec::with_capacity(NUM_USERS);
-    for _ in 0..NUM_USERS {
-        let records_for_user = generate_random_user_records_in_reverse_chronological_order(
+    while random_user_records.len() < NUM_USERS && total_count < MAX_QUERY_SIZE {
```

Comment:
Maybe you don’t need both now. Perhaps just a total number of records is sufficient. The “num users” won’t necessarily even matter if you pick the wrong combination of numbers.